client/export.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/cheggaaa/pb/v3"
+	"github.com/sapcc/go-bits/logg"
+	"github.com/sapcc/gophercloud-sapcc/v2/audit/v1/events"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
+)
+
+type ExportFormat string
+
+const (
+	ExportFormatJSON ExportFormat = "json"
+	ExportFormatYAML ExportFormat = "yaml"
+	ExportFormatCSV  ExportFormat = "csv"
+)
+
+var (
+	allExportFormats = []ExportFormat{ExportFormatJSON, ExportFormatYAML, ExportFormatCSV}
+)
+
+func ParseExportFormat(input string) (ExportFormat, error) {
+	if slices.Contains(allExportFormats, ExportFormat(input)) {
+		return ExportFormat(input), nil
+	}
+	return "", fmt.Errorf("unsupported format: %s (supported formats: %v)", input, allExportFormats)
+}
+
+// convertToRequestedFormat converts events to the specified format and writes to the provided buffer
+func convertToRequestedFormat(buf *bytes.Buffer, allEvents []events.Event, format string) error {
+	switch format {
+	case "json":
+		data, err := json.MarshalIndent(allEvents, "", "  ")
+		if err != nil {
+			return fmt.Errorf("failed to marshal JSON: %w", err)
+		}
+		buf.Write(data)
+
+	case "yaml":
+		data, err := yaml.Marshal(allEvents)
+		if err != nil {
+			return fmt.Errorf("failed to marshal YAML: %w", err)
+		}
+		buf.Write(data)
+
+	case "csv":
+		if err := writeCSV(buf, allEvents, defaultListKeyOrder); err != nil {
+			return fmt.Errorf("failed to write CSV: %w", err)
+		}
+
+	default:
+		return fmt.Errorf("unsupported format: %s", format)
+	}
+
+	return nil
+}
+
+// ExportCmd represents the export command
+var ExportCmd = &cobra.Command{
+	Use:   "export",
+	Args:  cobra.ExactArgs(0),
+	Short: "Export Hermes events to Swift",
+	Long: `Export Hermes events to Swift storage container.
+Exports can be saved in different formats (json, csv, yaml) for further processing or archival.`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if err := viper.BindPFlags(cmd.Flags()); err != nil {
+			return fmt.Errorf("failed to bind flags: %w", err)
+		}
+
+		// Validate container name is provided
+		if viper.GetString("container") == "" {
+			return errors.New("container name is required")
+		}
+
+		// Validate format
+		_, err := ParseExportFormat(viper.GetString("format"))
+		if err != nil {
+			return err
+		}
+
+		return verifyGlobalFlags(defaultListKeyOrder)
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := cmd.Context()
+
+		// Warn if trying to export more than default limit
+		if viper.GetInt("limit") > maxOffset {
+			fmt.Fprintf(os.Stderr, "Warning: Exporting more than %d events may take a long time.\n\n", maxOffset)
+		}
+
+		// Get events using existing list functionality
+		client, err := NewHermesV1Client(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create Hermes client: %w", err)
+		}
+
+		// Configure debug logging if enabled
+		if viper.GetBool("debug") {
+			transport := &http.Transport{}
+			client.HTTPClient = http.Client{
+				Transport: transport,
+			}
+		}
```

Comment:
does this code really enable the debug logging?

Reply:
Yes. I mean, I run it on command line and it provides information to help me debug. What am I missing from the question?

Reply:
Oh nevermind, I understand. I need to read a bit closer. You are correct.